capsules/src/l3gd20.rs

```diff
@@ -377,16 +408,19 @@ impl Driver for L3gd20Spi<'_> {
     fn subscribe(
         &self,
         subscribe_num: usize,
-        mut callback: Upcall,
-        _appid: ProcessId,
+        mut upcall: Upcall,
+        process_id: ProcessId,
     ) -> Result<Upcall, (Upcall, ErrorCode)> {
         match subscribe_num {
             0 /* set the one shot callback */ => {
-                callback = self.callback.replace (callback);
-                Ok (callback)
+                let _ = self
+                    .grants.enter(process_id, |grant| {
+                    mem::swap(&mut grant.upcall, &mut upcall);
+                });
+                Ok(upcall)
```

Comment:
This assumes entering the grant does not fail, but if entering the grant does fail, this will incorrectly return `Ok()` to the app despite the fact that the callback will never be called.

Reply:
Ah, right. I was operating under the assumption that one could only fail to enter the grant if the process has died (it therefore wouldn't matter if you returned Ok()). But it could also be that the grant region is exhausted. 

One note: I copied this code from the `spi_controller` capsule, so I suspect there are others that aren't handling this correctly.

Reply:
I copied this from `spi_controller`. If the only failure is a process fails, I think this is OK. But since it can also fail if the grant region is depleted, I agree, we want to pass the error to userspace.

Reply:
Good catch! I suspect I was irritated because of the fact that issuing a command will indeed release the reservation once entering a Grant fails. Will issue fixes accordingly.

Reply:
Fixed in #2577.